Information scripting

lukedirtwalker opened the pull request.

lukedirtwalker committed:
Add information scripting plugin

lukedirtwalker committed:
Initial work for scripting in python, expose AST

The AstApi_Generated.cpp file is only included if it exists. Otherwise
we
assure that the project still builds and runs fine.

lukedirtwalker committed:
Introduce InformationNode and Property for scripting

dimitar-asenov commented:
Ok, I had a look, and before things get horribly out of hand we should do the following:
- Talk about these initial commits and the basic setup. I want to understand everything as deeply as you do, at least at this point. Otherwise I fear that I'll be totally overwhelmed afterwards.
- Make sure that this is what we should be doing. At least so far, this python stuff looks a bit scary to me, and we should not forget that this is only a part of the project. I don't want the scripting to take a huge amount of time, and if it seems like it might, then we should reconsider and focus more on the parts that are more relevant and less dependent on external integrations.

lukedirtwalker committed:
Remove unused includes

lukedirtwalker committed:
Python linking for Windows and update doc for python.

This is not tested.

lukedirtwalker committed:
Move script to InformationScripting directory

lukedirtwalker committed:
Only use using namespace boost instead of boost::python

It makes it more explicit what comes from the boost::python world, e.g.
python::object makes it immediately clear what is meant.

lukedirtwalker committed:
Add conversion operator to Property

lukedirtwalker committed:
Introduce PropertyMap as a base class of InformationNode

The Edge in the graph will also use the same base class as they both can
have properties.

lukedirtwalker committed:
Use INFORMATIONSCRIPTING_API macro consistenly and fix indentation

lukedirtwalker commented on InformationScripting/InformationScripting.pro:
```diff
@@ -0,0 +1,56 @@
+}
+
+
+QT += qml
```
Remove!

lukedirtwalker committed:
Fix various small things in InformationScripting

-Remove qml module, its not needed.
-Use QList in PropertyMap as we will probably never have many
properties.
-Other small things as renamings.

lukedirtwalker commented:
Here as well I cleaned the issues we discussed. I decided to leave the Property as is currently but added a comment, so that we can reconsider this later.

Not that when we merge #107 first, this PR will conflict and vice versa, because of the envision.pro file.

lukedirtwalker committed:
Move Property definitions out of class

lukedirtwalker committed:
Remove invalid inline specifier and remove unimplemented method
